addons/stock/models/stock_move_line.py

```diff
@@ -743,37 +732,74 @@ def get_aggregated_properties(move_line=False, move=False):
+
+                    previous_move_lines = move_line.move_id.move_line_ids.filtered(
+                        lambda ml: self.get_aggregated_properties(move=ml.move_id)[0] == line_key and ml.id != move_line.id
+                    )
+                    qty_ordered -= sum(map(lambda m: m.product_uom_id._compute_quantity(m.qty_done, uom), previous_move_lines))
                 aggregated_move_lines[line_key] = {'name': name,
                                                    'description': description,
-                                                   'qty_done': move_line.qty_done,
-                                                   'qty_ordered': qty_ordered,
+                                                   'qty_done': qty_done,
+                                                   'qty_ordered': qty_ordered or qty_done,
                                                    'product_uom': uom.name,
                                                    'product_uom_rec': uom,
                                                    'product': move_line.product_id}
             else:
-                aggregated_move_lines[line_key]['qty_done'] += move_line.qty_done
+                aggregated_move_lines[line_key]['qty_ordered'] += qty_done
+                aggregated_move_lines[line_key]['qty_done'] += qty_done
 
         # Does the same for empty move line to retrieve the ordered qty. for partially done moves
         # (as they are splitted when the transfer is done and empty moves don't have move lines).
         if kwargs.get('strict'):
             return aggregated_move_lines
         pickings = (self.picking_id | backorders)
-        for empty_move in pickings.move_lines.filtered(
-            lambda m: m.state == "cancel" and m.product_uom_qty
-            and float_is_zero(m.quantity_done, precision_rounding=m.product_uom.rounding)
-        ):
-            line_key, name, description, uom = get_aggregated_properties(move=empty_move)
+        for empty_move in pickings.move_lines:
+            if not (empty_move.state == "cancel" and empty_move.product_uom_qty
+                    and float_is_zero(empty_move.quantity_done, precision_rounding=empty_move.product_uom.rounding)):
+                continue
+            line_key, name, description, uom = self.get_aggregated_properties(move=empty_move)
+
+            if line_key not in aggregated_move_lines:
+                qty_ordered = empty_move.product_uom_qty
+                aggregated_move_lines[line_key] = {
+                    'name': name,
+                    'description': description,
+                    'qty_done': False,
+                    'qty_ordered': qty_ordered,
+                    'product_uom': uom.name,
+                    'product': empty_move.product_id,
+                }
+            else:
+                aggregated_move_lines[line_key]['qty_ordered'] += empty_move.product_uom_qty
+
+        return aggregated_move_lines
+
+    def _get_empty_moves(self):
```

Comment:
So, if we have to keep `_get_empty_moves`, it should be cool to avoid to duplicate the code.
I suggest you to make a method called by both `_get_aggregated_product_quantities` and `_get_empty_moves`.
For example, in `stock_move.py`:
``` python3
    def _get_empty_moves(self, aggregated_move_lines=False):
        aggregated_move_lines = aggregated_move_lines or {}
        for empty_move in self:
            if empty_move.state != "cancel" or not empty_move.product_uom_qty or \
               not float_is_zero(empty_move.quantity_done, precision_rounding=empty_move.product_uom.rounding):
                continue
            line_key, name, description, uom = self.env['stock.move.line'].get_aggregated_properties(move=empty_move)

            if line_key not in aggregated_move_lines:
                qty_ordered = empty_move.product_uom_qty
                aggregated_move_lines[line_key] = {
                    'name': name,
                    'description': description,
                    'qty_done': False,
                    'qty_ordered': qty_ordered,
                    'product_uom': uom.name,
                    'product': empty_move.product_id,
                }
            else:
                aggregated_move_lines[line_key]['qty_ordered'] += empty_move.product_uom_qty
        return aggregated_move_lines
```

And then in `stock_move_line.py`:
```python3
    def _get_aggregated_product_quantities(self, **kwargs):
        ...
        pickings = (self.picking_id | backorders)
        return pickings.move_lines._get_empty_moves(aggregated_move_lines)

    def _get_empty_moves(self):
        backorders = self.env['stock.picking']
        pickings = self.picking_id
        while pickings.backorder_ids:
            backorders |= pickings.backorder_ids
            pickings = pickings.backorder_ids
        pickings = (self.picking_id | backorders)
        return pickings.move_lines._get_empty_moves()
```